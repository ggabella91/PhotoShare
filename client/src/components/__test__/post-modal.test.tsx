import { shallow } from 'enzyme';
import React from 'react';
import PostModal from '../post-modal/post-modal.component';

it('expect to render a post modal component', () => {
  const modalWrapper = shallow(
    <PostModal
      caption='hi'
      location='here'
      createdAt={new Date('Sat Jan 02 2021')}
      show={true}
      fileString='string'
      userName='Giuliano'
      onHide={() => {}}
      onOptionsClick={() => {}}
      userProfilePhotoFile='userFile'
    />
  );

  expect(modalWrapper).toMatchSnapshot();
});
