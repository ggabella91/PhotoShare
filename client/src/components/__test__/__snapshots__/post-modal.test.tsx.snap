// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render a post modal component 1`] = `
<Modal
  animation={false}
  autoFocus={true}
  backdrop={true}
  centered={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  dialogClassName="post-modal"
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
>
  <div
    className="large-image-adjustments"
  >
    <img
      alt="user"
      className="post-modal-image-large"
      src="data:image/jpeg;base64,string"
    />
  </div>
  <ModalHeader
    className="post-modal-header"
    closeButton={true}
    closeLabel="Close"
  />
  <ModalBody
    className="post-modal-body"
  >
    <img
      alt="post-pic"
      className="post-modal-image-embedded"
      src="data:image/jpeg;base64,string"
    />
    <div
      className="post-modal-details"
    >
      <div
        className="post-user-and-location"
      >
        <img
          alt="user"
          className="user-photo"
          src="data:image/jpeg;base64,userFile"
        />
        <div
          className="text-and-options"
        >
          <div
            className="user-and-location"
          >
            <span
              className="user-name"
            >
              Giuliano
            </span>
            <span
              className="post-location"
            >
              here
            </span>
          </div>
          <div
            className="post-options"
          >
            <span
              className="ellipsis"
              onClick={[Function]}
            >
              ...
            </span>
          </div>
        </div>
      </div>
      <span
        className="post-caption"
      >
        hi
      </span>
      <span
        className="post-date"
      >
        Thu Dec 24 2020
      </span>
    </div>
  </ModalBody>
</Modal>
`;
